org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -186,6 +187,7 @@ public class JDTLanguageServer extends BaseJDTLanguageServer implements Language
 	 */
 	private ServiceStatus status;
 	private TelemetryManager telemetryManager;
+	private Object codeAction = new Object();
```

Comment:
I would probably call it `codeActionLock`.

Reply:
Fixed